lib/topologies/mongos.js

```diff
@@ -1152,71 +1144,72 @@ Mongos.prototype.auth = function(mechanism, db) {
 
   // Topology is not connected, save the call in the provided store to be
   // Executed at some point when the handler deems it's reconnected
-  if (!self.isConnected() && self.s.disconnectHandler != null) {
-    return self.s.disconnectHandler.add('auth', db, allArgs, {}, callback);
+  if (!this.isConnected() && this.s.disconnectHandler != null) {
+    return this.s.disconnectHandler.add('auth', source, [credentials, callback], {}, callback);
   }
 
-  // Set to authenticating
-  this.authenticating = true;
-  // All errors
-  var errors = [];
-
   // Get all the servers
-  var servers = this.connectedProxies.slice(0);
+  const servers = this.connectedProxies.slice(0);
+
+  // Get total count
+  let count = servers.length;
+
   // No servers return
-  if (servers.length === 0) {
+  if (count === 0) {
     this.authenticating = false;
-    callback(null, true);
+    return callback(null, true);
```

Comment:
wow looks like that was a long outstanding bug, are you sure we haven't come to rely on this behavior?

Reply:
I really really hope not